packages/playwright-test/src/reporters/junit.ts

```diff
@@ -205,7 +293,7 @@ function escape(text: string, stripANSIControlSequences: boolean, isCharacterDat
   const escapeRe = isCharacterData ? /[&<]/g : /[&"<>]/g;
   text = text.replace(escapeRe, c => ({ '&': '&amp;', '"': '&quot;', '<': '&lt;', '>': '&gt;' }[c]!));
   if (isCharacterData)
-    text = text.replace(/]]>/g, ']]&gt;');
+    text = '<![CDATA[' + text.replace(/]]>/g, ']]&gt;') + ']]>';
```

Comment:
Well, I remember that whenever I was testing I wasn't getting some errors because it would produce invalid XML on the parser I was using. Nevertheless, if that block is character data it should be in a [CDATA block](https://www.w3.org/TR/REC-xml/#sec-cdata-sect). It doesn't make sense in my perspective, know that is character data and then not properly identify as such, right? :)
https://www.w3.org/TR/REC-xml/#sec-cdata-sect

Reply:
It's a bit unrelated indeed but I think it should be the standard behaviour (I don't see a need to create an option for it unless there's a known use case where having CDATA is going to break anything for any parser out there, which I doubt but sometimes there are basic parsers...)

Reply:
Yeah, I am hesitant changing it by default - just in case some of the parsers in the wild breaks and we regress existing users. An option is a safe bet, and we can recommend it in the docs for all new users. A separate PR with this option would be ideal 😄

Reply:
I've made a new PR  for that #12741